### PR TITLE
feat(core): add @coauthors/core to use only core interfaces in @coauthors/cli, precommit, coauthors.vercel.app

### PR DESCRIPTION
**Pull request by manudeli**
To clarify the dependencies of @coauthors/precommit and @coauthors/coauthors.vercel.app, in my opinion, we should separate essential logic by adding @coauthors/core.

After this PR merging, @coauthors/coauthors.vercel.app will use these interfaces to create page in coauthors.vercel.app/generator. (Page that serves the same role as the current https://coauthors.vercel.app)

image

### CODE REVIEW

File: biome.jsonc [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
+  "linter": {
+    "enabled": true,
+    "rules": {
+      "recommended": true
+    }
+  },
+  "formatter": {
+    "enabled": true,
+    "indentWidth": 2,
+    "indentStyle": "space"
```
Comment (manudeli) [inline biome.jsonc:15]: I update indent style, if you don't like you can request change. In my opinion, this change will make us read code in GitHub easily by short indent. like below picture

image

File: configs/tsup/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  target: [
+    "chrome51",
+    "firefox53",
+    "edge18",
+    "safari11",
+    "ios11",
+    "opera38",
+    "es6",
+    "node14",
+  ],
```
Comment (manudeli) [inline configs/tsup/src/index.ts:14]: We support lower version enviroments after this change

File: configs/tsup/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  entry: ["src/*.{ts,tsx}", "!**/*.{spec,test,test-d}.*"],
```
Comment (manudeli) [inline configs/tsup/src/index.ts:15]: We support multiple entry after this change

File: packages/cli/src/index.ts [hunk 4]
```diff
@@ -1,40 +1,18 @@
...
-	const coAuthors = await Promise.all(
-		authors.map(async ({ user, name }) => {
-			const resUser = await fetchGithubUser(user);
-			return `Co-authored-by: ${name ?? resUser.name} <${
-				resUser.id
-			}+${user}@users.noreply.github.com>`;
-		}),
-	);
-	return coAuthors.join("\n");
-}
```
Comment (manudeli) [inline packages/cli/src/index.ts:34]: I devide it as `coauthors` and `coauthor` of @coauthors/core

File: packages/cli/src/index.ts [hunk 5]
```diff
@@ -1,40 +1,18 @@
...
-
-async function fetchGithubUser(user: string) {
-	const res = await fetch(`https://api.github.com/users/${user}`);
-	if (!res.ok) throw new Error(`Failed to fetch user: ${res.statusText}`);
-	return await res.json();
-}
```
Comment (manudeli) [inline packages/cli/src/index.ts:40]: I renamed this as internal interface `githubUser` of @coauthors/core.

**Review by 2-NOW — APPROVED** [review]

**Pull request merged**